pkg/conversion/conversion.go

```diff
@@ -44,3 +56,48 @@ func UnmarshalData(from metav1.Object, to interface{}) (bool, error) {
 	from.SetAnnotations(annotations)
 	return true, nil
 }
+
+func UpperToPascal(str string) string {
+	if len(str) == 0 {
+		return str
+	}
+
+	// check for any exception in map
+	if exception, found := upperPascalExceptions[str]; found {
+		return exception
+	}
+
+	// Split on '-' or '_' rune, capitalize first letter of each part and join them
+	var sb strings.Builder
+	array := regexp.MustCompile(`[\-\_]+`).Split(strings.ToLower(str), -1)
```

Comment:
Perhaps pre-compute the regex? That's generally a good practice and won't change your design. Like this on package scope:

```go
var upperTokenizer = regexp.MustCompile(`[\-\_]+`)
```

(You may find me nitpicking but I think just a static map of that dozen of converted strings would have done the job without the need for any compute like this - it's called at every read/write of the CR, at least as long as v1beta1 is still the stored version. If they were many more strings to convert I'd agree that a map wouldn't be the best choice. Anyway that's not a huge compute and conversions don't happen so often so I'm fine with this option as well.)

Reply:
sure, done in https://github.com/netobserv/network-observability-operator/pull/483/commits/0d51d44b1ca04a35c5e5ef3e57942a6f46bf4add